### DISCUSSION TITLE
Connection Problem Loki Gateway

### DISCUSSION THREAD

**Discussion by micgruber** [serilog-contrib/serilog-sinks-grafana-loki#246] [Q&A] [answered]
I have a problem connecting to the loki gateway? I can't see any requests from the logging provider but when I run a simple http get request within the app I can access the loki gateway. Did I missed something for the logging configuration?

![problem](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/assets/19686895/8390d79c-c850-4545-8bb6-cbe27afa3ca8)

**Comment by mishamyte** [comment, marked as answer]
Try to enable `SelfLog` and check would any useful info be there or not.

[Doc here](https://github.com/serilog/serilog/wiki/Debugging-and-Diagnostics#selflog)

Plus the issue could be related to the lifecycle of the logger. Pls, check also [that doc link](https://github.com/serilog/serilog/wiki/Lifecycle-of-Loggers)

**Comment by micgruber** [comment]
Thanks, I found the issue you have to set at least one label:

![image](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/assets/19686895/061a696d-9783-42a4-b890-abf9934d4e90)

One thing which I had to add was a flush to see the error:

`Log.CloseAndFlush();`

**Reply by mishamyte** [reply]
Glad to hear you solved the issue!

**Comment by Aashay-Madavi** [comment]
what uri you guys are using to wite the logs in garafana loki @micgruber and @mishamyte

**Reply by mishamyte** [reply]
Huh?

It completely depends on your system configuration. You can have Loki deployed locally in a cluster in certain ways or use the cloud one, for example

**Reply by Aashay-Madavi** [reply]
i have deployed loki in kubernetes and having problem to do so, and i also want to ask where can i see the selflog logs in kubernetes i m not able to get that too it would be very kind if you help @mishamyte

**Reply by mishamyte** [reply]
> i have deployed loki in kubernetes and having problem to do so, and i also want to ask where can i see the selflog logs in kubernetes i m not able to get that too it would be very kind if you help @mishamyte 

Pls, check the official documentation on how `SelfLog` works (in typical case, it redirects to stdout)

**Reply by JulioMoreyra** [reply]
Same problem here. Selflog shows 404 
Watching the logs in the server i see the uri and its wrong because the library duplicates the path

![Screenshot_20241220_115257](https://github.com/user-attachments/assets/eb666ad7-8bf3-44e2-9795-14e86b41e7a2)

So,i pointed the uri to the root of the k8s gw and it worked.
Hope that helps
